nautobot/extras/querysets.py

```diff
@@ -127,10 +127,11 @@ def _get_config_context_filters(self):
                 (Q(device_redundancy_groups=OuterRef("device_redundancy_group")) | Q(device_redundancy_groups=None)),
                 Q.AND,
             )
-            base_query.add((Q(locations=OuterRef("location")) | Q(locations=None)), Q.AND)
-
+            # This is necessary to prevent location related config context to be applied now.
```

Comment:
```suggestion
            # This is necessary to prevent location related config context to be applied now. The location hierarchy cannot be processed by the database and must be added by `ConfigContextModel.get_config_context`
```